pkgs/development/libraries/jpeg-xl/default.nix

```diff
@@ -0,0 +1,69 @@
+{ stdenv, lib, fetchgit
+, cmake
+, giflib
+, gperftools
+, gtest
+, libjpeg
+, libpng
+, libwebp
+, openexr
+, pkgconfig
+, zlib
+}:
+
+stdenv.mkDerivation rec {
+
+  pname = "jpeg-xl";
+  version = "2020-10-29-8f2e0857";
+
+  src = fetchgit { # as of writing, `fetchFromGitLab` cannot fetch submodules
+    # Temporary workaround for https://gitlab.com/wg1/jpeg-xl/-/issues/57
+    url = "https://gitlab.com/nh21/jpeg-xl";
+    # url = "https://gitlab.com/wg1/jpeg-xl";
+    rev = "8f2e08572e686b8016e1efac0a5eb5bc7fca1b81";
+    sha256 = "0i58vr1jmawma4jck3v2ym76adksknj8jasnxvfj96zs2q5y025i";
+    # There are various submodules in `third_party/`.
+    fetchSubmodules = true;
+    # The `CMakeLists.txt` errors on absence of `.git` directories.
+    leaveDotGit = true;
+  };
+
+  nativeBuildInputs = [
+    cmake
+    gtest
+    pkgconfig
+  ];
+
+  # Functionality not currently provided by this package
+  # that the cmake build can apparently use:
+  #     OpenGL
+  #     GLUT
+  #     Doxygen
+
+  # As of writing, jpeg-xl does not point out all its dependencies
+  # conclusively in its README or otherwise; they can best be determined
+  # by checking the CMake output for "Could NOT find".
+  buildInputs = [
+    giflib
+    gperftools # provides `libtcmalloc`
+    libjpeg
+    libpng
+    libwebp
+    openexr
+    zlib
+  ];
+
+  cmakeFlags = [
+  ];
+
+  doInstallCheck = true;
+  installCheckTarget = "test";
+
+  meta = with lib; {
+    homepage = http://libjpeg-turbo.virtualgl.org/;
```

Comment:
```suggestion
    homepage = "https://libjpeg-turbo.virtualgl.org/";
```

Reply:
Oops, I forgot to put in jpeg-xl's URL overall.

Reply:
Done.